### PR TITLE
EVG-2471 Add new ec2 provider to UI

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cloud/new_ec2.go [hunk 1]
```diff
@@ -68,10 +68,23 @@ func (s *NewEC2ProviderSettings) Validate() error {
 	return nil
 }
 
+type providerType int
```
Comment (tychoish) [inline cloud/new_ec2.go]: shouldn't this be ec2ProviderType?

File: cloud/new_ec2.go [hunk 2]
```diff
@@ -263,12 +276,12 @@ func (m *ec2Manager) SpawnHost(h *host.Host) (*host.Host, error) {
 	defer m.client.Close()
 
 	var resources []*string
-	if m.onDemand(h) {
+	if m.provider == onDemandProvider {
```
Comment (tychoish) [inline cloud/new_ec2.go]: can we delete the onDemand/spot predicate methods?

File: cloud/new_ec2.go (hunk 1 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

File: cloud/new_ec2.go (hunk 2 above)
Comment (bsamek) [reply cloud/new_ec2.go]: We cannot. We don't use them for `SpawnHost` anymore, but for `GetInstanceStatus` and `CostForDuration` we need to find out the type of the host. Once we implement the auto spawn type, we will not know ahead of time, for a given host, whether we spawned it as on-demand or spot.

Comment (tychoish) [reply cloud/new_ec2.go]: then shouldn't we continue to use these methods to avoid having multiple ways of determining instance type.

lgtm otherwise.

Comment (bsamek) [reply cloud/new_ec2.go]: We need to answer two different questions:
# Should I spawn an on-demand or a spot host?
# Is this host on-demand or spot?

These are different questions. One is a feature of the provider and will change over time. The other is a function of an (already-existing host). I renamed the functions to `isHostSpot` and `isHostOnDemand` for clarity.

**Pull request merged**